### PR TITLE
Navigation book

### PR DESCRIPTION
**Pull request by EstebanIndiveri**
## Summary

Add React-navigation Stack

## Screenshots
<a href="https://ibb.co/c6Jy7FY">screenshot-1620164445460</a>
Android 9

<a href="https://ibb.co/FW9cb6P">pixel2</a>
<a href="https://ibb.co/dcYTCTZ">Screenshot-1620229589</a>
Google Pixel 2

<a href="https://ibb.co/6RgKyK1">galaxy4-7</a>
Galaxy nexus 4.7

### iOS

<a href="https://ibb.co/42tgTQ6">ios8</a>
<a href="https://ibb.co/WkjwGvL">ios8null</a>
Iphone 8

<a href="https://ibb.co/bJsSr59">Phone-SE-2nd-generation-2021-05-05-at-09-31-40</a>
<a href="https://ibb.co/wrn6FGg">Simulator-Screen-Shot-i-Phone-SE-2nd-generation-2021-05-05-at-12-48-58</a>
Iphone SE 2

<a href="https://ibb.co/ZHY2PWp">navigation2ios</a>
<a href="https://ibb.co/198hCNy">Simulator-Screen-Shot-i-Phone-12-2021-05-05-at-12-48-27</a>
## Trello Card
[Trello card Link Navigation](https://trello.com/c/8LM05JxT/13-n-71-introducci%C3%B3n-a-react-navigation)

### CODE REVIEW

File: wbooks/src/app/components/CardBook/index.tsx [hunk 1]
```diff
@@ -11,19 +13,22 @@ interface Props {
...
-      <Image
-        source={image ? { uri: image } : imageNotFound}
-        style={styles.image}
-        resizeMode={image ? 'contain' : 'cover'}
-        resizeMethod="resize"
-      />
-      <View style={styles.containerText}>
-        <Text style={styles.titleCard}>{title}</Text>
-        <Text style={styles.authorCard}>{author}</Text>
+    <TouchableWithoutFeedback onPress={() => navigation.navigate('DetailBook', { image, title, author, id })}>
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/components/CardBook/index.tsx]: create a const and include this

File: wbooks/src/app/screens/DetailBook/styles.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    fontSize: 30
+  },
+  paragraph: {
+    marginTop: 10,
+    textAlign: 'center'
+  },
+  images: {
+    height: 400,
+    borderBottomRightRadius: 20,
+    borderBottomLeftRadius: 20
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/screens/DetailBook/styles.js]: use a const to same values

File: wbooks/src/app/components/CardBook/index.tsx [hunk 3]
```diff
@@ -11,19 +13,25 @@ interface Props {
   id: number;
 }
 
-export default function CardBook({ image, title, author }: Props): ReactElement {
+export default function CardBook({ image, title, author, id }: Props): ReactElement {
+  const navigation = useNavigation();
+  const handleClick = () => {
+    navigation.navigate('DetailBook', { image, title, author, id });
```
Comment (mattgle) [inline wbooks/src/app/components/CardBook/index.tsx]: you can have a routes.ts file that contains all the Routes names from screens. U can import it then and use it here.

File: wbooks/src/app/screens/DetailBook/styles.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Platform, StyleSheet } from 'react-native';
+import { whiteBlue } from '@constants/colors';
+
+const borderRadiusBottom = 20;
+const platformMarginTop = Platform.OS === 'ios' ? 60 : 20;
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/screens/DetailBook/styles.js]: to const use uppercase: `const BORDER_RADIUS_BOTTON`

File: wbooks/src/app/components/CardBook/index.tsx [hunk 5]
```diff
@@ -11,19 +13,25 @@ interface Props {
...
-      <Image
-        source={image ? { uri: image } : imageNotFound}
-        style={styles.image}
-        resizeMode={image ? 'contain' : 'cover'}
-        resizeMethod="resize"
-      />
-      <View style={styles.containerText}>
-        <Text style={styles.titleCard}>{title}</Text>
-        <Text style={styles.authorCard}>{author}</Text>
+    <TouchableWithoutFeedback onPress={handleClick}>
```
Comment (mattgle) [inline wbooks/src/app/components/CardBook/index.tsx]:
```suggestion
    <TouchableWithoutFeedback onPress={handlePress}>
```

File: wbooks/src/app/components/CardBook/index.tsx [hunk 6]
```diff
@@ -11,19 +13,25 @@ interface Props {
   id: number;
 }
 
-export default function CardBook({ image, title, author }: Props): ReactElement {
+export default function CardBook({ image, title, author, id }: Props): ReactElement {
+  const navigation = useNavigation();
+  const handleClick = () => {
```
Comment (mattgle) [inline wbooks/src/app/components/CardBook/index.tsx]:
```suggestion
  const handlePress = () => {
```

File: wbooks/src/app/navigation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+export default function Navigation(): ReactElement {
+  return (
+    <Stack.Navigator
+      headerMode="float"
+      screenOptions={{
+        cardStyleInterpolator: CardStyleInterpolators.forHorizontalIOS
+      }}
+      initialRouteName="Home">
+      <Stack.Screen
+        name="Home"
```
Comment (mattgle) [inline wbooks/src/app/navigation/index.tsx]: You can use ROutes file here

File: wbooks/src/app/navigation/index.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        component={Library}
+        options={{
+          title: 'Library',
+          headerTitleAlign: 'center'
+        }}
+      />
+      <Stack.Screen
+        name="DetailBook"
```
Comment (mattgle) [inline wbooks/src/app/navigation/index.tsx]: You can use ROutes file here

File: wbooks/src/app/navigation/index.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import React, { ReactElement } from 'react';
+import { CardStyleInterpolators, createStackNavigator } from '@react-navigation/stack';
+import Library from '@screens/Library';
+import DetailBook from '@screens/DetailBook';
+import { white } from '@constants/colors';
+
+const Stack = createStackNavigator();
+
+export default function Navigation(): ReactElement {
```
Comment (mattgle) [inline wbooks/src/app/navigation/index.tsx]:
```suggestion
export default function Navigator(): ReactElement {
```

File: wbooks/src/app/screens/DetailBook/index.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+export default function DetailBook({
+  route: {
+    params: { author, title, image }
+  }
+}: Props): ReactElement {
+  return (
+    <View style={styles.container}>
+      {image && (
+        <Image source={{ uri: image }} style={styles.images} resizeMethod="scale" resizeMode="cover" />
+      )}
```
Comment (andreszuluaga-wolox) [inline wbooks/src/app/screens/DetailBook/index.tsx]: add view with backgroundColor when image is null

**Review by mattgle — CHANGES_REQUESTED** [review]

File: wbooks/src/app/routes/index.tsx [hunk 11]
```diff
@@ -0,0 +1,2 @@
+export const DETAIL_BOOK = 'DetailBook';
+export const HOME = 'Home';
```
Comment (mattgle) [inline wbooks/src/app/routes/index.tsx]:
```suggestion
export default enum Routes {
DetailBook = 'DetailBook',
Home = 'Home'
}
```

Comment (mattgle) [reply wbooks/src/app/routes/index.tsx]: The file should be called routes.ts and it MUST be at `src/constants`

**Review by mattgle — CHANGES_REQUESTED** [review]

**Review by mattgle — APPROVED** [review]

**Pull request merged**